### PR TITLE
Include currency definition in sales, purchases, and cases db models

### LINKED ISSUE #1826: Include a currency definition in the order transactions api responses
`/ob/sales`, `/ob/purchases` and `/ob/cases` return a `total` with each transaction. However, no currency definition is returned. To be consistent with other APIs which include a cur def any time we have a currency amount, please include a cur def for each transaction.

### PR DESCRIPTION
**Pull request by cpacia**
Currently these models represent a total as a string. This commit changes it
to be a CurrencyValue to match the other APIs which return a CurrencyValue.

Tests are updated to reflect this change.

Closes #1826

### CODE REVIEW

**Comment by jjeffryes** [comment]
@cpacia which APIs will change as a result of this? I assume the sales, purchases, and cases calls?

Any others?

@rmisio FYI.

**Comment by rmisio** [comment]
@jjeffryes I'm aware. I logged the bug this PR closes. It's a pretty quick change for the client to adjust.

File: repo/db/cases.go [hunk 1]
```diff
@@ -283,13 +287,22 @@ func (c *CasesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sortByA
 			}
 		}
 
+		cur, err := repo.AllCurrencies().Lookup(paymentCoin)
+		if err != nil {
+			return nil, 0, err
+		}
+		cv, err := repo.NewCurrencyValue(total.String(), cur)
```
Comment (placer14) [inline repo/db/cases.go]: Both calls can be done together with `repo.NewCurrencyValueWithLookup`

File: repo/db/purchases.go [hunk 2]
```diff
@@ -193,13 +197,22 @@ func (p *PurchasesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sor
 			coinType = ""
 		}
 
+		cur, err := repo.AllCurrencies().Lookup(paymentCoin)
+		if err != nil {
+			return nil, 0, err
+		}
+		cv, err := repo.NewCurrencyValue(totalStr, cur)
```
Comment (placer14) [inline repo/db/purchases.go]: Both calls can be done together with `repo.NewCurrencyValueWithLookup`

File: repo/db/sales.go [hunk 3]
```diff
@@ -193,13 +198,22 @@ func (s *SalesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sortByA
 			coinType = ""
 		}
 
+		cur, err := repo.AllCurrencies().Lookup(paymentCoin)
+		if err != nil {
+			return nil, 0, err
+		}
+		cv, err := repo.NewCurrencyValue(totalStr, cur)
```
Comment (placer14) [inline repo/db/sales.go]: Both calls can be done together with `repo.NewCurrencyValueWithLookup`

File: repo/db/util.go [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
 	if contract.BuyerOrder.Payment.Coin != "" {
-		return contract.BuyerOrder.Payment.Coin
+		return contract.BuyerOrder.Payment.Coin, nil
 	}
 	if len(contract.VendorListings[0].Metadata.AcceptedCurrencies) > 0 {
-		return contract.VendorListings[0].Metadata.AcceptedCurrencies[0]
+		return contract.VendorListings[0].Metadata.AcceptedCurrencies[0], nil
 	}
-	return ""
+	return "", errors.New("payment coin not foun")
```
Comment (placer14) [inline repo/db/util.go]: Typo: "found"

**Pull request merged**